File: RebrewsViewModels/ViewModels/Recipe/RecipeViewModel.cs
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace RebrewsViewModels.ViewModels.Recipe
{
    [DataContract]
    public class RecipeViewModel
    {
        [DataMember]
        public int Id { get; set; }

        [DataMember]
        public string Name { get; set; }



        [DataMember]
        public ICollection<RecipeFermentableViewModel> Fermentables { get; set; }

        [DataMember]
        public ICollection<RecipeHopViewModel> Hops { get; set; }

        [DataMember]
        public ICollection<RecipeYeastViewModel> Yeasts { get; set; }


    }
}

File: RebrewsWeb/typings/RecipeViewModel.ts
module Rebrews { 
    export interface RecipeViewModel { 
        id: number;
        name: string;
        fermentables: RecipeFermentableViewModel[];
        hops: RecipeHopViewModel[];
        yeasts: RecipeYeastViewModel[];
        profile: RecipeProfileViewModel;
        style: RecipeStyleViewModel;
        style_Id: number;
    }
}